libraries/ms-common/src/x86/crc32.c

```diff
@@ -60,3 +61,18 @@ uint32_t crc32_arr(const uint8_t *buffer, size_t buffer_len) {
   // Invert the result
   return ~crc & 0xffffffff;
 }
+
+uint32_t crc32_append_arr(const uint8_t *buffer, size_t buffer_len, uint32_t initial_crc) {
```

Comment:
Same here: looks like you can just replace `crc32_arr` with a call to `crc32_append_arr(buffer, buffer_len, 0)`. (Maybe you can do the same for the stm32 version too once it's tested.)